src/migrations/20230420125500-v5-strategy-changes.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+exports.up = function (db, callback) {
+    db.runSql(
+        `
+        -- delete deprecated strategies still present in v4
+        delete from strategies 
+            where name in ('gradualRolloutUserId', 'gradualRolloutRandom', 'gradualRolloutSessionId') 
+            and deprecated 
+            and not exists (select * from feature_strategies where strategy_name = name limit 1);
```

Comment:
This migration only deletes the strategies if they match the happy path. As we discussed there are two blockers here for deleting them:
1. the strategy is in use
2. the strategy is in use and it has been modified.

For 1. we can probably migrate the use into using gradualRollout with the same parameters as before.
For 2. we can change the name and make it clear that its handled by the user and not unleash anymore just like other custom strategies.

Reply:
This is a very good comment. Ideally, we'd like to get rid of them, but there might be unexpected side effects that make me worry.

> For 1. we can probably migrate the use into using gradualRollout with the same parameters as before.

Yes, but we need to merge 3 strategies into one, which might be doable in some situations, but still won't cover all cases. I'm also not that sure how stickiness will behave if we change the strategy, because even if the strategy implementation is the same, because of the name change we might reassign users to another treatment. So I rather keep it like this